### PR TITLE
Bug 1181069 - Add code transformer for Raptor tests

### PR DESCRIPTION
**Pull request by snowmantw**
APP=system DEBUGUY=<where the debuguy is> TESTDIR=<where the esp files are> make

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=929903871beea629c43d875a3ea3723e5b61d765

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e1bad10eef42195a2533c89b1edd7c5e643fa4c0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7e031dd1a7d04db9610a262fa7b9cd99c7f41301

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=67ad1476797b7578ff4712bcba951a772897b695

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=39e8f1478b4dd27fb8f773d91d23bec658568b49

File: Makefile [hunk 1]
```diff
@@ -557,7 +557,9 @@ define BUILD_CONFIG
   "PERF_LOGGING" : "$(PERF_LOGGING)", \
   "SHARE_PERF_USAGE": "$(SHARE_PERF_USAGE)", \
   "DEFAULT_KEYBOAD_SYMBOLS_FONT": "$(DEFAULT_KEYBOAD_SYMBOLS_FONT)", \
-  "DEFAULT_GAIA_ICONS_FONT": "$(DEFAULT_GAIA_ICONS_FONT)" \
+  "DEFAULT_GAIA_ICONS_FONT": "$(DEFAULT_GAIA_ICONS_FONT)", \
+	"RAPTOR_TRANSFORM": "$(RAPTOR_TRANSFORM)", \
```
Comment (rickychien) [inline Makefile]: nit: add space

File: build/raptor-test-transformer.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * 3. Check if such file exists
+ * 4. Run this transformer + espect to apply the esp file on to target file
+ */
+var utils = require('utils');
+
+exports.execute = function(options) {
+  var webapp = options.webapp;
+  var debuguypath = options.RAPTOR_TRANSFORMER_PATH;
+  if (!debuguypath) {
+    debuguypath = options.GAIA_DIR + '/node_modules/debuguy';
```
Comment (rickychien) [inline build/raptor-test-transformer.js]: this script will be executed in xpcshell environment so I recommend that handle path by utils.joinPath() for better compatible with windows.

File: build/raptor-test-transformer.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  var file = utils.getFile(debuguypath);
+  if (!file.exists()) {
+    throw new Error('Cannot find debuguy: set the path of tool as DEBUGUY');
+  }
+  var debuguyclipath = debuguypath + '/cli.js';
```
Comment (rickychien) [inline build/raptor-test-transformer.js]: same

File: build/raptor-test-transformer.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  var node = new utils.Commander('node');
+  node.initPath(utils.getEnvPath());
+  node.run([debuguyclipath, 'autolog', '--x-espect=' +
+    options.GAIA_DIR + '/build/raptor-test-transformer.esp.js' , '--r',
```
Comment (rickychien) [inline build/raptor-test-transformer.js]: same

File: .jshintignore [hunk 5]
```diff
@@ -8,6 +8,7 @@ apps/calendar/js/ext/**
 apps/camera/js/vendor/**
 apps/camera/test/**
 apps/privacy-panel/js/vendor/**
+apps/**/*.esp.js
```
Comment (timdream) [inline .jshintignore]: Why? Are we putting generated code into source dir?

Comment (snowmantw) [inline .jshintignore]: No: `*.esp` is the template file you put in your test directory of the app, which instructs what code (ex: `performance.mark`) should be woven into the code in `build_stage` and the profile. See the demo I put in this patch: ` apps/system/lockscreen/raptor/lockscreen.esp.js`.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b66e9309d313131c584ee9f54eecabd313e5da71

**Pull request merged**